File: app/src/main/java/denis/easyweather/app/common/BaseFragment.java
package denis.easyweather.app.common;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import denis.easyweather.app.inject.components.ActivityComponent;
import denis.easyweather.app.router.MainActivity;
import denis.easyweather.app.router.Router;

/**
 * Created by Denis on 06-Feb-17.
 */

public abstract class BaseFragment extends Fragment {

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        inject();
    }

    protected Router getRouter() {
        return ((MainActivity) getActivity()).getRouter();
    }

    protected abstract void inject();

    protected ActivityComponent getActivityComponent() {
        return ((MainActivity)getActivity()).getActivityComponent();
    }
}

File: app/src/main/java/denis/easyweather/app/common/BaseFragment.kt
package denis.easyweather.app.common

import android.os.Bundle
import android.support.v4.app.Fragment

import denis.easyweather.app.inject.components.ActivityComponent
import denis.easyweather.app.router.MainActivity
import denis.easyweather.app.router.Router

abstract class BaseFragment : Fragment() {

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        inject()
    }

    protected val router: Router?
        get() = (activity as MainActivity).router

    protected abstract fun inject()

    protected val activityComponent: ActivityComponent?
        get() = (activity as MainActivity).activityComponent
}
